Skinned Cube/MinecraftSkin.cs

```diff
@@ -30,51 +30,24 @@ public class MinecraftSkin {
-                var request = WebRequest.Create(String.Format(SKIN_URL_NICKNAME, playername));
-                var buffer = new byte[4096];
-                var skin = new BitmapImage();
-                using (var target = new MemoryStream()) {
-                    try {
-                        using (var response = (HttpWebResponse)request.GetResponse()) {
-                            using (var stream = response.GetResponseStream()) {
-                                int read;
-
-                                while (stream != null && (read = stream.Read(buffer, 0, buffer.Length)) > 0) {
-                                    target.Write(buffer, 0, read);
-                                }
-                            }
-
-                            skin.BeginInit();
-                            skin.CacheOption = BitmapCacheOption.OnLoad;
-                            skin.StreamSource = target;
-                            skin.EndInit();
-                        }
-                    }
-                    catch {
-                        //web exception, or maybe decoding the image failed
-                        return Steve;
-                    }
-                }
-
-                skin.Freeze();
-                return new MinecraftSkin(skin);
-            });
-            return mcSkin;
-        }
 
-        private static async Task<String> GetSkinUrl(String uuid) {
-            using (var client = new WebClient()) {
-                var rawJson = await client.DownloadStringTaskAsync(new Uri("https://sessionserver.mojang.com/session/minecraft/profile/" + uuid.Replace("-", String.Empty)));
-                dynamic json = DynamicJson.Parse(rawJson);
-                var properties = ((dynamic[])json.properties).First(p => p.name == "textures").value;
+            return await Task.Run(async () => {
+                                      try {
+                                          using (var client = new WebClient()) {
+                                              var rawJson = client.DownloadString(new Uri("https://api.mojang.com/users/profiles/minecraft/" + playername));
+                                              dynamic json = DynamicJson.Parse(rawJson);
+                                              return await LoadByUuidAsync(json["id"]);
+                                          }
+                                      }
+                                      catch (Exception) {
+                                          return Steve;
+                                      }
+                                  });
 
-                var textureData = DynamicJson.Parse(System.Text.Encoding.UTF8.GetString(Convert.FromBase64String(properties)));
-                var textureUrl = textureData.textures.SKIN.url;
-                return textureUrl;
-            }
         }
 
+        private static async Task<String> GetSkinUrl(String uuid) { return "https://crafatar.com/skins/" + uuid.Replace("-", String.Empty); }
```

Comment:
This should better be implemented without an external service, e.g. by using [this Minecraft API method](https://wiki.vg/Mojang_API#UUID_-.3E_Profile_.2B_Skin.2FCape).